nix/versions.nix

```diff
@@ -0,0 +1,11 @@
+{lib}: let
+  ## It’s much easier to read from a JSON file than to have JSON import from some other file, so we extract some
+  ## configuration from the VS Code settings to avoid duplication.
+  vscodeSettings = lib.importJSON ../.vscode/settings.json;
+in
+  vscodeSettings."haskell.toolchain"
+  ## There are some things we want to pin that the VS Code Haskell extension doesn’t let us control.
+  // {
+    hpack = "0.35.2";
```

Comment:
I have seen some hpack 0.37 slipping in, so we may want to bump this too at some point.
https://github.com/unisonweb/unison/blob/754805ca1638cb726870510f8ad7555ac4abb5ab/unison-share-api/unison-share-api.cabal#L3

Reply:
Yeah, good call. In #5486, I mention that 0.36 is the minimum supported by our pinned stack version, so it’s even inconsistent within the current Nix environment. I should rebase that, now that this is merged, and see if the cache issue is gone.